src/Modules/Data/Commands.Data/SetPowerBIDataset.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Management.Automation;
+using Microsoft.PowerBI.Common.Api.Datasets;
+using Microsoft.PowerBI.Common.Api.Workspaces;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Data
+{
+    [Cmdlet(CmdletVerb, CmdletName, DefaultParameterSetName = DatasetIdParameterSetName)]
+    public class SetPowerBIDataset : PowerBIClientCmdlet
+    {
+        public const string CmdletVerb = VerbsCommon.Set;
+        public const string CmdletName = "PowerBIDataset";
+
+        #region ParameterSets
+        private const string DatasetIdParameterSetName = "DatasetId";
+        private const string DatasetNameParameterSetName = "DatasetName";
+        private const string ObjectIdParameterSetName = "ObjectAndId";
+        private const string ObjectNameParameterSetName = "ObjectAndName";
+        #endregion
+
+        #region Parameters
+
+        [Parameter(Mandatory = true, ParameterSetName = DatasetIdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName)]
+        public Guid Id { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DatasetNameParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = DatasetIdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = DatasetNameParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectIdParameterSetName)]
+        [Parameter(Mandatory = true, ParameterSetName = ObjectNameParameterSetName)]
+        [ValidateSet("Abf", "PremiumFiles")]
```

Comment:
What is Abf? Might need to spell out and then convert unless this is well documented

Reply:
Checked -- it's ok